### ISSUE TITLE
Queries sometimes hang for an additional minute

### ISSUE THREAD

**Issue by teskje** [MaterializeInc/materialize#17227]
### What version of Materialize are you using?

main

### How did you install Materialize?

Materialize Cloud

### What is the issue?

Interactive queries that usually complete fairly quickly sometimes instead take an additional minute to complete. The cause of this seem to be timeouts during fetching blobs from S3 in persist.

Specifically, I noticed this while running TPC-H benchmarks. It is possible that this issue occurs more when large amounts of data are involved.

### Reproduction

I could only reproduce this in cloud (staging), not locally. It might reproduce locally as well, if persist was configured to use S3 for the blob store.

Run the following queries to set up the TPC-H sources and a test cluster:
```
CREATE SOURCE lgtpch FROM LOAD GENERATOR TPCH (SCALE FACTOR 5, TICK INTERVAL '1s') FOR ALL TABLES WITH (SIZE = 'xsmall');

CREATE CLUSTER test REPLICAS (r1 (SIZE 'large'));
SET cluster = test;
```

Now wait a bit until the TPC-H sources have caught up.

Define a view for TPC-H query 2. Others will probably work as well.

```
CREATE VIEW q02 AS SELECT
    s_acctbal,
    s_name,
    n_name,
    p_partkey,
    p_mfgr,
    s_address,
    s_phone,
    s_comment
FROM
    part, supplier, partsupp, nation, region
WHERE
    p_partkey = ps_partkey
    AND s_suppkey = ps_suppkey
    AND p_size = CAST (15 AS smallint)
    AND p_type LIKE '%BRASS'
    AND s_nationkey = n_nationkey
    AND n_regionkey = r_regionkey
    AND r_name = 'EUROPE'
    AND ps_supplycost
        = (
                SELECT
                    min(ps_supplycost)
                FROM
                    partsupp, supplier, nation, region
                WHERE
                    p_partkey = ps_partkey
                    AND s_suppkey = ps_suppkey
                    AND s_nationkey = n_nationkey
                    AND n_regionkey = r_regionkey
                    AND r_name = 'EUROPE'
            )
ORDER BY
    s_acctbal DESC, n_name, s_name, p_partkey;
```

Repeatedly run the query until the described issue occurs:
```
materialize=> \timing
Timing is on.
materialize=> \o /dev/null
materialize=> SELECT * FROM q02;
Time: 11262.504 ms (00:11.263)
materialize=> SELECT * FROM q02;
Time: 11282.394 ms (00:11.282)
materialize=> SELECT * FROM q02;
Time: 65270.738 ms (01:05.271)
```

### Initial Investigation

I have traced this down to blob fetching from S3. In particular, if you add logging around this part of the code:
https://github.com/MaterializeInc/materialize/blob/845f8ed2aca78f86638053a106cc56d5da98d411/src/persist-client/src/fetch.rs#L236-L240

... you see that it blocks for about one minute:

```
│ clusterd {"timestamp":"2023-01-19T14:42:09.610269Z","level":"INFO","fields":{"message":"[...] fetching..."},"target":"mz_persist_client::fetch"}          │
│ clusterd {"timestamp":"2023-01-19T14:43:10.804087Z","level":"INFO","fields":{"message":"[...] fetching... done"},"target":"mz_persist_client::fetch"}     │
```

That one minute suspiciously matches [the read timeout we configure with the S3 SDK](https://github.com/MaterializeInc/materialize/blob/845f8ed2aca78f86638053a106cc56d5da98d411/src/persist/src/s3.rs#L105). So a likely explanation for the above behavior is that reading the blob from S3 fails for some time, times our after 60 seconds, and immediately succeeds on the second try.

Relevant discussion [in Slack](https://materializeinc.slack.com/archives/C01CFKM1QRF/p1674140180968459).

**Comment by aljoscha** [comment]
@pH14 should I assign this to you? It's about the timeouts in S3 that we are looking at, right?

**Comment by teskje** [comment]
I opened a PR with Paul's suggested easy fix of reducing the S3 read timeout: https://github.com/MaterializeInc/materialize/pull/17239.

**Comment by teskje** [comment]
Fixed in #17246.
